src/events/mod.rs

```diff
@@ -138,52 +138,33 @@ impl<'a> BytesStart<'a> {
         }
     }
 
-    /// Creates a new `BytesStart` from the given content (name + attributes).
+    /// Creates a new `BytesStart` from the given name.
     ///
     /// # Warning
     ///
-    /// `&content[..name_len]` is not checked to be a valid name
+    /// `name` is not checked to be a valid name
     #[inline]
-    pub fn borrowed(content: &'a str, name_len: usize) -> Self {
+    pub fn new<C: Into<Cow<'a, str>>>(name: C) -> Self {
+        let buf = str_cow_to_bytes(name);
         BytesStart {
-            buf: Cow::Borrowed(content.as_bytes()),
-            name_len,
+            name_len: buf.len(),
+            buf,
         }
     }
 
-    /// Creates a new `BytesStart` from the given name.
+    /// Creates a new `BytesStart` from the given content (name + attributes).
     ///
     /// # Warning
     ///
-    /// `name` is not checked to be a valid name
-    #[inline]
-    pub fn borrowed_name(name: &'a str) -> BytesStart<'a> {
-        Self::borrowed(name, name.len())
-    }
-
-    /// Creates a new `BytesStart` from the given content (name + attributes)
-    ///
-    /// Owns its contents.
+    /// `&content[..name_len]` is not checked to be a valid name
```

Comment:
The attributes are also not checked to be valid.

Reply:
To be honest I'm not sure I see the use case for this function for the majority of users, ironically the one use which is clear is simplifying integration tests by making it easy to create events from invalid data.

I'm not saying that we should remove it, but perhaps it can be discouraged more actively?

Reply:
Actually, we should mark that functions `unsafe` (#153) and provide a safe versions returning `Result` and which will check for a validity. But first we need to implement that validity checks (#371 and others) :). So I think it is a task for another PR, probably the one that will add these checks

Reply:
Sure, sounds fine.  Let's still do the documentation part though.

Reply:
Could you write a suggestion?

Reply:
How about 

> `&content[..name_len]` must be a valid name, and the remainder of `content` must be valid attributes. Neither are checked, it is possible to generate invalid XML if `content` or `name_len` are incorrect.

Reply:
Done + added similar warnings to all other constructors where there were none